### PR TITLE
guides: Creating a PrivateLink to TriggerMesh Services

### LINKED ISSUE #296: Guide: PrivateLink to Knative Services

### PR DESCRIPTION
**Pull request by sameersbn**
Closes #296

### CODE REVIEW

File: docs/guides/knative-privatelink.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+[PrivateLink](https://aws.amazon.com/privatelink) is a service that enables you to establish private connectivity 
+between VPCs and services hosted on Amazon Web Services (AWS) or on-premises, 
+without exposing data to the internet. Let’s look at the configuration required
+to access Knative services deployed on an AWS Elastic Kubernetes Service (AWS) 
+cluster across AWS accounts over a PrivateLink connection.
+
+## Configuring the Istio Ingress Gateway
+
+In the default configuration, the Istio networking layer for Knative creates an
```
Comment (jmcx) [inline docs/guides/knative-privatelink.md:11]: Maybe worth mention something like 

"the Istio networking layer (or other such as Contour or Kourrier)"

**Review by jmcx — APPROVED** [review]
Minor comments, amazing quality post!

File: docs/guides/triggermesh-privatelink.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+After the configuration has been applied successfully, we can see that the Istio 
+Ingress Gateway is now using an NLB as shown in the screenshot below:
+
+![NLB](../assets/guides/privatelink/nlb.png)
+
+## Setting up the PrivateLink endpoint service
+
+Before we set up the PrivateLink, let’s make sure that the VPC to which our EKS 
+cluster is attached has the DNS hostnames feature enabled. This is needed in order 
+to be able to access our Knative services using the URL configured for the services.
```
Comment (jmcx) [inline docs/guides/triggermesh-privatelink.md:48]: knative still mentioned here, maybe not problem

File: docs/guides/triggermesh-privatelink.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+[PrivateLink](https://aws.amazon.com/privatelink) is a service that enables you to establish private connectivity 
+between VPCs and services hosted on Amazon Web Services (AWS) or on-premises, 
+without exposing data to the internet. Let’s look at the configuration required
+to access TriggerMesh services deployed on an AWS Elastic Kubernetes Service (AWS) 
+cluster across AWS accounts over a PrivateLink connection.
+
+## Configuring the Istio Ingress Gateway
+
+The Istio networking layer (or others such as Kourier) creates an Elastic Load 
+Balancer (ELB) allowing public access to the Knative services running in the cluster.
```
Comment (jmcx) [inline docs/guides/triggermesh-privatelink.md:12]: knative still mentioned here, maybe not problem

File: docs/guides/triggermesh-privatelink.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+names which is available under the Modify private DNS name action menu item of 
+the endpoint connection as shown below:
+
+![Modify Private DNS Name](../assets/guides/privatelink/modify-private-dns-name.png)
+
+## Accessing TriggerMesh services
+
+To access the TriggerMesh services via the PrivateLink, we can create an EC2 instance, 
+in the consumers AWS account, on the same VPC in which the endpoint was created 
+and use cURL to reach the Knative service endpoints.
```
Comment (jmcx) [inline docs/guides/triggermesh-privatelink.md:125]: knative still mentioned here, maybe not problem

**Review by jmcx — APPROVED** [review]

File: docs/guides/triggermesh-privatelink.md (hunk 3 above)
Comment (sameersbn) [reply docs/guides/triggermesh-privatelink.md:12]: 👍

**Pull request merged**